src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -88,53 +83,82 @@ public static void Init(ISuppressionFileSystemService fileSystemService, IEnumer
 
 		public static void SaveSuppressionBase()
 		{
-			if (Instance == null)
+			CheckIfInstanceIsInitialized();
+			
+			lock (Instance._fileSystemService)
 			{
-				throw new InvalidOperationException($"{nameof(SuppressionManager)} instance was not initialized");
-			}
+				//Create local copy in order to avoid concurency problem when the collection is changed during the iteration
+				var filesWithGeneratedSuppression = Instance._fileByAssembly.Values.Where(f => f.GenerateSuppressionBase).ToList();
 
-			var filesWithGeneratedSuppression = Instance._fileByAssembly.Values.Where(f => f.GenerateSuppressionBase);
-
-			foreach (var file in filesWithGeneratedSuppression)
-			{
-				Instance._fileSystemService.Save(file.MessagesToDocument(), file.Path);
+				foreach (var file in filesWithGeneratedSuppression)
+				{
+					Instance._fileSystemService.Save(file.MessagesToDocument(), file.Path);
+				}
 			}
 		}
 
-		private bool IsSuppressed(SemanticModel semanticModel, Diagnostic diagnostic, CancellationToken cancellation)
+		public static SuppressionFile CreateSuppressionFileForProject(Project project)
 		{
-			cancellation.ThrowIfCancellationRequested();
+			project.ThrowOnNull(nameof(project));
+			CheckIfInstanceIsInitialized();
 
-			var (assembly, message) = GetSuppressionInfo(semanticModel, diagnostic, cancellation);
+			//First check if file already exists to dismiss threads withou acquiring the lock
+			if (Instance._fileByAssembly.TryGetValue(project.Name, out var existingSuppressionFile) && existingSuppressionFile != null)
+				return existingSuppressionFile;
 
-			if (assembly == null)
+			lock (Instance._fileSystemService)
 			{
-				return false;
+				//Second check inside the lock if file already exists 
+				Instance._fileByAssembly.TryGetValue(project.Name, out existingSuppressionFile);
+				return existingSuppressionFile ?? AddNewSuppressionFileImpl();
 			}
 
-			var file = _fileByAssembly.GetOrAdd(assembly, (SuppressionFile)null);
-
-			if (file == null)
+			//---------------------------------------------Local Function--------------------------------------------------
+			SuppressionFile AddNewSuppressionFileImpl()
 			{
-				return false;
-			}
+				string suppressionFileName = project.Name + SuppressionFile.SuppressionFileExtension;
+				string projectDir = Instance._fileSystemService.GetFileDirectory(project.FilePath);
+				string suppressionFilePath = Path.Combine(projectDir, suppressionFileName);
 
-			if (file.GenerateSuppressionBase)
-			{
-                if (diagnostic?.Descriptor.DefaultSeverity != DiagnosticSeverity.Info)
-                {
-                    file.AddMessage(message);
-                }
+				//Create new xml document and get its text
+				System.Xml.Linq.XDocument newXDocument = SuppressionFile.NewDocumentFromMessages(Enumerable.Empty<SuppressMessage>());
```

Comment:
Why not to add the System.Xml.Linq namespace to usings?

Reply:
It's just one place - didn't want to litter Intellisense

Reply:
But it is not as readable as it could be. Maybe it is better to use "var" instead of "System.Xml.Linq.XDocument"?

Reply:
ok. 
```suggestion
				var newXDocument = SuppressionFile.NewDocumentFromMessages(Enumerable.Empty<SuppressMessage>());
```